graylog2-server/src/main/java/org/graylog2/storage/versionprobe/VersionProbe.java

```diff
@@ -75,9 +77,14 @@ public VersionProbe(ObjectMapper objectMapper,
         this.delayBetweenAttempts = elasticsearchVersionProbeDelay;
         this.isJwtAuthentication = runsWithDataNode || opensearchUseJwtAuthentication;
         this.indexerJwtAuthTokenProvider = indexerJwtAuthTokenProvider;
+        this.loggingListener = new VersionProbeLogger(LOG);
```

Comment:
is there a reason why you have that as a parameter for each function? it is initialized as a field here anyway and could be used directly if you introduce a second constructor where you can pass a custom one in.

Reply:
I wanted to make it obvious which logger is used. For the It makes sense to keep the same logger as before (preserving backwards compatibility for log4j configuration). But for DatanodeConnectivityCheck there is a different logger used. So each usage has different parameter. 

Using a default-initialized VersionProbe logger inside the VersionProbeLogger (mismatch of class and logger) feels like a bug or something we want to avoid and trigger warnings in code analysis tools.

Reply:
ok, but I would prefer to remove the field completely then to avoid misunderstandings

Reply:
Thanks, fixed!